compose/service.py

```diff
@@ -769,9 +770,9 @@ def build(self, no_cache=False, pull=False, force_rm=False):
 
         build_opts = self.options.get('build', {})
         path = build_opts.get('context')
-        # python2 os.path() doesn't support unicode, so we need to encode it to
-        # a byte string
-        if not six.PY3:
+        # python2 os.stat() doesn't support unicode on some UNIX, so we
+        # encode it to a bytestring to be safe
+        if not six.PY3 and not IS_WINDOWS_PLATFORM:
```

Comment:
Are you sure this fixes the issue? Because I'm using Python 3 and the source from Git directly and still having this issue.